instant-xml-macros/src/lib.rs

```diff
@@ -2,62 +2,129 @@ extern crate proc_macro;
 
 use proc_macro::TokenStream;
 use quote::quote;
-use syn::{parse_macro_input, DeriveInput, Lit, Meta, NestedMeta};
+use std::collections::HashMap;
+use syn::{parse_macro_input, Lit, Meta, NestedMeta};
 
-fn retrieve_default_namespace(input: &DeriveInput) -> Option<String> {
-    for attr in &input.attrs {
-        if !attr.path.is_ident(XML) {
-            continue;
-        }
+const XML: &str = "xml";
+
+struct Serializer {
+    default_namespace: Option<String>,
+    other_namespaces: HashMap<String, String>,
+}
+
+impl<'a> Serializer {
+    pub fn init(attributes: &'a Vec<syn::Attribute>) -> Serializer {
+        let mut default_namespace: Option<String> = None;
+        let mut other_namespaces = HashMap::<String, String>::default();
+
+        for attr in attributes {
+            if !attr.path.is_ident(XML) {
+                continue;
+            }
 
-        let nested = match attr.parse_meta() {
-            Ok(Meta::List(meta)) => meta.nested,
-            Ok(_) => todo!(),
-            _ => todo!(),
-        };
+            let nested = match attr.parse_meta() {
+                Ok(Meta::List(meta)) => meta.nested,
+                Ok(_) => todo!(),
+                _ => todo!(),
+            };
+
+            let list = match nested.first() {
+                Some(NestedMeta::Meta(Meta::List(list))) => list,
+                _ => todo!(),
+            };
 
-        let list = match nested.first() {
-            Some(NestedMeta::Meta(Meta::List(list))) => list,
-            _ => todo!(),
-        };
+            if let Some(ident) = list.path.get_ident() {
```

Comment:
For this case, it would be better to use a `match` block so that you can do an early return if `get_ident()` yields `None` (or if `ident != "namespace"`).